runtime/jcl/common/jcldefine.c

```diff
@@ -29,7 +29,7 @@
 
 jclass 
 defineClassCommon(JNIEnv *env, jobject classLoaderObject,
-	jstring className, jbyteArray classRep, jint offset, jint length, jobject protectionDomain, UDATA options, J9Class *hostClass, J9ClassPatchMap *patchMap)
+	jstring className, jbyteArray classRep, jint offset, jint length, jobject protectionDomain, UDATA * options, J9Class *hostClass, J9ClassPatchMap *patchMap, UDATA allowedBitsForClassName)
```

Comment:
It doesn't look like the modified options value is used by any caller, so there's really no need to pass this by pointer. You are free to modify the options value directly.

Reply:
the options was checked by `Java_java_lang_ClassLoader_defineClassImpl` as follows:
```
Java_java_lang_ClassLoader_defineClassImpl(JNIEnv *env, jobject receiver, jstring className, jbyteArray classRep, jint offset, jint length, jobject protectionDomain)
{
....
	jclass result = defineClassCommon(env, receiver, className, classRep, offset, length, protectionDomain, &options, NULL, NULL, CLASSNAME_VALID);

	if (J9_ARE_ANY_BITS_SET(options, J9_FINDCLASS_FLAG_NAME_IS_INVALID) <-----------
	&& (NULL == result) && (NULL == currentThread->currentException)) {
		/*
		 * Now that we know the class is not exempt, throw NoClassDefFoundError as we
		 * would have liked to have done above before we called defineClassCommon().
		 */
		vmFuncs->internalEnterVMFromJNI(currentThread);
		vmFuncs->setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGNOCLASSDEFFOUNDERROR, (UDATA *)*(j9object_t *)className);
		vmFuncs->internalExitVMToJNI(currentThread);
	}
```
We can't move the setup of exception into `defineClassCommon` as other callers won't set up the NoClassDefFoundError exception.

Reply:
Sorry, missed that.